.github/workflows/ios.yml

```diff
@@ -0,0 +1,37 @@
+name: iOS CI
+
+on:
+  push:
+    branches: [ master]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  build:
+    name: Build and Test default scheme using any available iPhone simulator
+    runs-on: macos-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Relisten
+        run: |
+          git clone --quiet https://github.com/alecgorge/gapless-audio-bass-ios.git ../BASSGaplessAudioPlayer
+          git clone --quiet https://github.com/alecgorge/AGAudioPlayer.git ../AGAudioPlayer
+          pod update RealmSwift
+          ./setup.sh
+          scheme_list=$(xcodebuild -list -json | tr -d "\n")
+          default=$(echo $scheme_list | ruby -e "require 'json'; puts JSON.parse(STDIN.gets)['project']['targets'][0]")
+          echo $default | cat >default
+          echo Using default scheme: $default
+      - name: Build
+        env:
+          scheme: ${{ 'default' }}
+          platform: ${{ 'iOS Simulator' }}
+        run: |
+          device=`instruments -s -devices | grep -oE 'iPhone.*?[^\(]+' | head -1 | awk '{$1=$1;print}'`
+          if [ $scheme = default ]; then scheme=$(cat default); fi
+          if [ "`ls -A | grep -i \\.xcworkspace\$`" ]; then filetype_parameter="workspace" && file_to_build="`ls -A | grep -i \\.xcworkspace\$`"; else filetype_parameter="project" && file_to_build="`ls -A | grep -i \\.xcodeproj\$`"; fi
+          file_to_build=`echo $file_to_build | awk '{$1=$1;print}'`
+          xcodebuild build-for-testing -scheme "$scheme" -"$filetype_parameter" "$file_to_build" -destination "platform=$platform,name=$device"
```

Comment:
Why not use the `xcodebuild.sh` script here? It uses `xcpretty` to make the build output easier to read, and it also means we have a shared build script that works for both CI and developer's local machines.

The device detection script part is good- let's move that into `xcodebuild.sh` so that we automatically find a simulator device to use.

However, `instruments` is deprecated, and we'd be better off asking the simulator itself for a list of its devices:

`xcrun simctl list devices available iPhone | sed -nE 's/^ +(iPhone[^\(]*) \(.*/\1/gp' | tail -n1`

Reply:
Nice catch